### ISSUE TITLE
Next major Laragon update?

### ISSUE THREAD

**Issue by jorqensen** [leokhoa/laragon#225]
I'm wondering when we will see the next "major" update for Laragon? I personally like Laragon for the simplicity and ease of use, however, with PHP 8 released & MySQL v8 the current download for Laragon seems quite outdated. It's very easy to update everything yourself, but slightly defeats the purpose. Furthermore, the PHP v8 update seems to break the extensions unless you want to edit the files manually, will this be fixed anytime soon?

I would love to see Laragon get a 2021 update with all software up-to-date out of the box and less "bugs" regarding the settings.

**Some feedback:**

1. The app interface itself could be a little bit prettier, right now it's just text thrown in, something that looked a little more like XAMPP would drastically improve the professional feel. I wouldn't mind throwing a mock-up idea together of how it could look if you would like that. Feel free to reach out.

2. Will we ever see a Laragon download that just have PHP, MySQL, Nginx/Apache, Redis etc. and not other software such as HeidiSQL shipped with it? I am using JetBrains DataGrip and PhpStorm / VS Code, I have never used any of the tools shipping with Laragon, they're just there.

Thanks!

**Comment by jorqensen** [comment]
@leokhoa Sorry for the direct ping, I saw you reacted to the issue, care to share your thoughts or ideas behind the continued development of Laragon? There's no official blog/update list for Laragon, so some insight would be appreciated. 😊

**Comment by leokhoa** [comment]
@jorqensen : Here are my answers:
1. Personally, I don't like the XAMPP interface. Actually, you don't need to interact frequently with Laragon interface. Changing it will make other articles, videos people made outdated.
2. Laragon is portable. You can even download the minimal Laragon zip file from git:
https://github.com/leokhoa/laragon/archive/master.zip
Then, you can add your favorite packages from **Menu > Tools > Quick add**

For me, disks are cheap so if you don't use the pre-packaged tools, you can just ignore them or delete them.

I am developing a new killer feature for Laragon so I don't have much time so please understand if I don't reply quick enough.

**Comment by jorqensen** [comment]
@leokhoa : Understandable, I was mostly curious if you were slowly considering Laragon dead / discontinued or if we would be seeing an update sometime in the near future, the interface/shipped apps are not a problem to me, just an idea. 👍

Do you perhaps have any idea when we will see the next Laragon update, maybe move towards GitHub releases instead for smaller updates at a time rather than major releases? 🤔

**Comment by eru123** [comment]
@leokhoa : Please commit & push those development so we can track the development of Laragon, there are tons of developer waiting for the next Laragon release that can help on you find bugs and issues or to develop Laragon fast.

**Comment by NamesMT** [comment]
Sorry for bugging you out...
Hope the next major would also include a fix for #197, too, detect and make "laravel new" works like "composer create-project" does?, Hope you'll get through all the hard-time developing the software and releasing it soon.
Cảm ơn for bringing Laragon to us!

**Comment by jorqensen** [comment]
> Sorry for bugging you out...
> Hope the next major would also include a fix for #197, too, detect and make "laravel new" works like "composer create-project" does?, Hope you'll get through all the hard-time developing the software and releasing it soon.
> Cảm ơn for bringing Laragon to us!

`laravel new` is executed from the Laravel installer, you can just install this globally via. Composer, it shouldn't really be an issue to begin with IMO. 🙂

**Comment by NamesMT** [comment]
> `laravel new` is executed from the Laravel installer, you can just install this globally via. Composer, it shouldn't really be an issue to begin with IMO.

Yes I can install it and manually create a new laravel project with `laravel new`, but the issue is calling it with Laragon **Menu -> Quick app** (after modifying usr/sites.conf) doesn't work, it parses laravel as an archives and therefore opens up the download&extracting windows, then gives error, rather than opening the command prompt as with `composer create-project`  
Tried circumventing with composer -V & laragon new, didn't spark :'(.  
TL;DR: seems like you didn't read #197

**Comment by jorqensen** [comment]
> > `laravel new` is executed from the Laravel installer, you can just install this globally via. Composer, it shouldn't really be an issue to begin with IMO.
> 
> Yes I can install it and manually create a new laravel project with `laravel new`, but the issue is calling it with Laragon **Menu -> Quick app** (after modifying usr/sites.conf) doesn't work, it parses laravel as an archives and therefore opens up the download&extracting windows, then gives error, rather than opening the command prompt as with `composer create-project`
> Tried circumventing with composer -V & laragon new, didn't spark :'(.
> TL;DR: seems like you didn't read #197

Sorry, I misinterpreted issue #197 and assumed it was just getting `laravel new` to work overall and not from the config files. I stand corrected.

**Comment by anburocky3** [comment]
**Feature ideas from me:**
- Auto-update of Laragon app
- Should have the ability to update the packages in GUI version without leaving laragon dashboard.
- It would be great if people can customize the laragon apps (as @jorqensen  mentioned, many developers won't need all tools and thus, it is keeping the disk dusty. Manually deleting the files is not a solution. Laragon is to automate & reduce the developer's time.)

**We all know that you have lots of work to do:** It would be also great, if you made Laragon as open-source code, so the community can help others and keep the laragon running & catching with the packages.

**Comment by mnovello** [comment]
Is Laragon receiving any updates?

**Comment by jorqensen** [comment]
As this issue is slowly reaching it's 2 year mark, I have given up on tools like Laragon.

I understand the author is a busy man and I have no problem with that, but seeing as Laragon is a direct competitor to tools like XAMMP I would have assumed there would be more activity on the project.

I've switched to using [Scoop.sh](https://scoop.sh/) for dealing with development directly on my Windows machine. Install the tools you need like the popular package manager [Homebrew](https://brew.sh/).

It requires a bit more configuration and patience, but having my tools updated with a single command and frequently (because the buckets are maintained by the community) is a win for me.

I'll be writing up a guide on how to use it for daily development soon enough, so hopefully other Windows developers can get something out of it.

**Comment by leokhoa** [comment]
Thanks @jorqensen
I can't spend my time on things that doesn't bring much values to Laragon users.
You always have choices.

**Comment by anburocky3** [comment]
@leokhoa May i ask you something. If you don't have much time to work with features. Why can't you make this open-source, so others might give support and maintain this for longer period?

**Comment by mnovello** [comment]
I agree with what others have suggested here. It would be great if Laragon was open-sourced.

**Comment by leokhoa** [comment]
@anburocky3 : Security is my top priority to protect Laragon users.
The core of Laragon is stable and Laragon is extensible. Most of the time, you can get you want easily with Laragon without changing code and updating the binary.
For example: https://forum.laragon.org/topic/2753/a-laragon-portable-packages-configuration-file-kit

**Comment by anburocky3** [comment]
> @anburocky3 : Security is my top priority to protect Laragon users. The core of Laragon is stable and Laragon is extensible. Most of the time, you can get you want easily with Laragon without changing code and updating the binary. For example: https://forum.laragon.org/topic/2753/a-laragon-portable-packages-configuration-file-kit

We totally love **Laragon** just becoz of this extensible. Till now, i use Laragon for my projects and using `PHP 8.1`, `Latest MySQL` etc.

But it would be great, if we can do it in single action.

**Comment by jorqensen** [comment]
> @anburocky3 : Security is my top priority to protect Laragon users. The core of Laragon is stable and Laragon is extensible. Most of the time, you can get you want easily with Laragon without changing code and updating the binary. For example: https://forum.laragon.org/topic/2753/a-laragon-portable-packages-configuration-file-kit

I don't see how open-sourcing Laragon will make it less secure?

The main point is also that, even though Laragon is "easy to customize", I should not be downloading a binary in 2022 and be greeted with a soon-to-be EOL PHP version etc.

There are a lot of things that could be improved in Laragon. And unless you're actively maintaining the project, I don't see a point in _**not**_ making it open-source.

**Comment by anburocky3** [comment]
I too feel like that. Making open-source will definitely increase security. Becoz. Every code is transparent and @leokhoa will be the person who decides what comes in and out.

Checkout all other open source tools, its the huge plus of open sourcing.

**Comment by alxndr-w** [comment]
Don't get me wrong here - I'm not a supporter of "security by obscurity", but as @leokhoa announced and delivered a new version of Laragon, the original issue from 2021 is fortunately outdated now.

Meanwhile The [Wiki section](https://github.com/leokhoa/laragon/wiki/Miscellaneous) was heavily improved and updating/upgrading PHP or MySQL is an unchallenging task, I think.

**Comment by anburocky3** [comment]
> Don't get me wrong here - I'm not a supporter of "security by obscurity", but as @leokhoa announced and delivered a new version of Laragon, the original issue from 2021 is fortunately outdated now.
> 
> Meanwhile The [Wiki section](https://github.com/leokhoa/laragon/wiki/Miscellaneous) was heavily improved and updating/upgrading PHP or MySQL is an unchallenging task, I think.

@alxndr-w  My sincere congrats for being as contributor of Laragon project (I read from the announcement). But making this as outdated, doesn't solve the issue. 

Atleast the laragon team should consider the opinions of various developers. I taught more than 1 Million users around the world in YouTube, I have suggested them to use **Laragon** at that time.

Since that much users are using that project, making it good would be great. 

What we are asking is too little:
1. A script that will automates the process of downloading and upgrading the packages. 

> #### IMO, Telling everything to do it by ourselves(coding side), is not a great solution. Those are workarounds.
